tests/base/src/test/java/org/wildfly/security/http/client/ElytronHttpClientTest.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.http.client;
+
+import org.junit.Assert;
+import org.junit.Test;
+import static org.wildfly.security.http.HttpConstants.CONFIG_REALM;
+
+import org.wildfly.security.auth.client.AuthenticationConfiguration;
+import org.wildfly.security.auth.client.AuthenticationContext;
+import org.wildfly.security.auth.client.AuthenticationContextConfigurationClient;
+import org.wildfly.security.auth.client.ElytronXmlParser;
+import org.wildfly.security.auth.client.InvalidAuthenticationConfigurationException;
+import org.wildfly.security.auth.client.MatchRule;
+import org.wildfly.security.digest.WildFlyElytronDigestProvider;
+import org.wildfly.security.http.HttpServerAuthenticationMechanism;
+import org.wildfly.security.http.HttpServerAuthenticationMechanismFactory;
+import org.wildfly.security.http.basic.BasicMechanismFactory;
+import org.wildfly.security.http.digest.DigestMechanismFactory;
+import org.wildfly.security.http.impl.AbstractBaseHttpTest;
+import org.wildfly.security.password.WildFlyElytronPasswordProvider;
+
+import java.net.URL;
+import java.net.http.HttpRequest;
+import java.security.PrivilegedAction;
+import java.security.Provider;
+import java.security.Security;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static java.security.AccessController.doPrivileged;
+import static org.wildfly.security.http.HttpConstants.FORBIDDEN;
+import static org.wildfly.security.http.HttpConstants.UNAUTHORIZED;
+
+/**
+ * Test for the ElytronHttpClient class
+ *
+ * @author <a href="mailto:kekumar@redhat.com">Keshav Kumar</a>
+ */
+public class ElytronHttpClientTest extends AbstractBaseHttpTest {
```

Comment:
@keshav-725 Just a minor, please divide this test class into multiple test classes for clarity. Like ElytronHttpClientBasicTest, ElytronHttpClientDigestTest, ElytronHttpClientBearerTest, ElytronHttpClientTLSTest